tests/postgres_tests/test_serialfield.py

```diff
@@ -0,0 +1,71 @@
+from django.contrib.postgres.fields.serial import (
+    BigSerialField, SerialField, SmallSerialField,
+)
+from django.db import models
+from django.db.backends.postgresql.base import (
+    DatabaseWrapper as PostgresDatabaseWrapper,
+)
+from django.test import skipUnlessDBFeature
+from django.test.utils import isolate_apps
+
+from . import PostgreSQLTestCase
+from .models import SerialModel
+
+
+@isolate_apps('invalid_models_tests')
+class SerialFieldModelTests(PostgreSQLTestCase):
+    def test_not_unique(self):
+        class Model(models.Model):
+            id = SerialField(unique=False)
+
+        field = Model._meta.get_field('id')
+        errors = field.check()
+        expected = []
+        self.assertEqual(errors, expected)
+
+    def test_primary_key(self):
+        class Model(models.Model):
+            another = SerialField()
+
+        field = Model._meta.get_field('id')
+        errors = field.check()
+        expected = []
+        self.assertEqual(errors, expected)
+
+    def test_internal_type(self):
+        class Model(models.Model):
+            small = SmallSerialField()
+            regular = SerialField()
+            big = BigSerialField(primary_key=False)
+
+        small = Model._meta.get_field('small')
+        regular = Model._meta.get_field('regular')
+        big = Model._meta.get_field('big')
+
+        self.assertIn(regular.get_internal_type(), PostgresDatabaseWrapper.data_types)
+        self.assertIn(small.get_internal_type(), PostgresDatabaseWrapper.data_types)
+        self.assertIn(big.get_internal_type(), PostgresDatabaseWrapper.data_types)
+
+        self.assertEqual(PostgresDatabaseWrapper.data_types[regular.get_internal_type()], 'serial')
+        self.assertEqual(PostgresDatabaseWrapper.data_types[small.get_internal_type()], 'smallserial')
+        self.assertEqual(PostgresDatabaseWrapper.data_types[big.get_internal_type()], 'bigserial')
+
+
+@skipUnlessDBFeature('has_native_serial_field')
+class SerialFieldTests(PostgreSQLTestCase):
```

Comment:
I would suggest subclassing `IntegerFieldTests` to ensure that tests for validators, etc. are run for these fields too.

See changes to `tests/model_fields/test_autofield.py` in outstanding PR #10680 for an example.

This test should also be duplicated for a `BigSerialModel` and `SmallSerialModel`.

Reply:
I can't import that unit test, since it's from another app and imports models that are not loaded :/

Reply:
Ah, yes. I guess the tests mentioned will just need to be copied...